dask_expr/_dispatch.py

```diff
@@ -0,0 +1,5 @@
+from __future__ import annotations
+
+from dask.utils import Dispatch
+
+get_collection_type = Dispatch("get_collection_type")
```

Comment:
I don't think we will need many dispatch functions here. So, we could also define this in `dask.datafame.dispatch`. For example, we could call it `get_expr_collection_class` in `dask.datafame` to avoid confusion.

Reply:
This is now publicly exposed as `dask_expr.get_collection_type`. This can always be moved to `dask.datafame.dispatch` in the future.